### PR TITLE
Xcode 10 support

### PR DESCRIPTION
**Pull request by ikesyo**
#### Checklist
- [x] Updated CHANGELOG.md.

### CODE REVIEW

File: Sources/ValidatingProperty.swift [hunk 1]
```diff
@@ -178,26 +178,6 @@ public final class ValidatingProperty<Value, ValidationError: Swift.Error>: Muta
...
-	/// The proposed value is only committed when `valid` is returned by the
-	/// `validator` closure.
-	///
-	/// - note: `inner` is retained by the created property.
-	///
-	/// - parameters:
-	///   - inner: The inner property which validated values are committed to.
-	///   - other: The property that `validator` depends on.
-	///   - validator: The closure to invoke for any proposed value to `self`.
-	public convenience init<U, E>(
```
Comment (ikesyo) [inline Sources/ValidatingProperty.swift]: The capture of the compiler error

Comment (ikesyo) [reply Sources/ValidatingProperty.swift]: The following overload will be used instead after this removal: https://github.com/ReactiveCocoa/ReactiveSwift/blob/7c25bc2e284ba54a8d81d3843a59cdd933c8f823/Sources/ValidatingProperty.swift#L128-L132

Comment (sharplet) [reply Sources/ValidatingProperty.swift]: I think this makes sense 👍

File: Tests/ReactiveSwiftTests/SignalProducerSpec.swift [hunk 2]
```diff
@@ -2791,7 +2791,7 @@ class SignalProducerSpec: QuickSpec {
 					let producer = SignalProducer<Int, NoError>.never
 						.on(disposed: { disposed = true })
 
-					var replayedProducer = ImplicitlyUnwrappedOptional(producer.replayLazily(upTo: 1))
+					var replayedProducer = Optional(producer.replayLazily(upTo: 1))
```
Comment (sharplet) [inline Tests/ReactiveSwiftTests/SignalProducerSpec.swift]: So we were using it as a regular optional anyway? Huh.

**Comment by mdiep** [comment]
Do you want to move the headers phase before the compile sources phase as part of this?

**Comment by ikesyo** [comment]
> Do you want to move the headers phase before the compile sources phase as part of this?

Addressed in eb2bae1!

File: Sources/ValidatingProperty.swift (hunk 1 above)
Comment (andersio) [inline Sources/ValidatingProperty.swift]: This changes the semantic. [This overload](https://github.com/ReactiveCocoa/ReactiveSwift/pull/644/files/e27310de692416a4fd70edfd1896ff3245ca4b51#diff-42b514003f72de821971eab375eda51dR192) was supposed to be the primary one, and the removed one here should have been implemented in terms of that.

Comment (ikesyo) [reply Sources/ValidatingProperty.swift]: So what should be done here to avoid the compilation error? @andersio

Comment (ikesyo) [reply Sources/ValidatingProperty.swift]: Feel free pushing your changes on this branch.

**Pull request merged**